src/test/java/org/apache/commons/mail/resolver/DataSourceFileResolverTest.java

```diff
@@ -43,13 +45,14 @@ public void testResolvingFileLenient() throws Exception
         assertNull(dataSourceResolver.resolve("./images/does-not-exist.gif"));
     }
 
-    @Test(expected = IOException.class)
+    @Test
     public void testResolvingFileNonLenient() throws Exception
     {
         final DataSourceResolver dataSourceResolver = new DataSourceFileResolver(new File("."), false);
         assertNotNull(dataSourceResolver.resolve("./src/test/resources/images/asf_logo_wide.gif"));
 
-        dataSourceResolver.resolve("asf_logo_wide.gif");
+        final Executable testMethod = () -> dataSourceResolver.resolve("asf_logo_wide.gif");
+        assertThrows(IOException.class, testMethod);
```

Comment:
Hello @nhojpatrick 
Thank you for your PR.
As with the other PRs, the local variable is superfluous. 
Please rebase on master to fix the spotbugs build on Java 11 and 17.